utils/browser.py

```diff
@@ -77,18 +77,18 @@ def checkin(self):
             self.docker_id = None
 
     def renew(self):
-        # If we have a docker id, renew_timer shouldn't still be None
-        if self.docker_id and not self.renew_timer.is_alive():
-            # You can call renew as frequently as desired, but it'll only run if
-            # the renewal timer has stopped or failed to renew
-            response = self._get('renew', self.docker_id)
-            try:
-                expiry_info = json.loads(response.content)
-            except ValueError:
-                raise ValueError("JSON could not be decoded:\n{}".format(response.content))
-            self.config.update(expiry_info)
-            self._reset_renewal_timer()
-            log.info('Renewed webdriver container %s', self.docker_id)
+        assert self.docker_id, 'The Wharf.renew was called before Wharf.checkout.'
```

Comment:
So having looked at it closer, I do concur that because the ```.renew()``` method is being called inside the ```Timer``` thread, the ```is_alive()``` method is reporting that it is alive when in fact it is not. However I don't think we shuld assert here, we risk breaking things when it's called early, the if was probably safer.